src/app/pages/client-review/review.component.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { NgxSpinnerModule } from 'ngx-spinner';
+import { ReactiveFormsModule, FormsModule, NgForm } from '@angular/forms';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ToastrModule } from 'ngx-toastr';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { throwError, of, Subject } from 'rxjs';
+import { Router, ActivatedRoute } from '@angular/router';
+import { routerSpy, toastServiceSpy, spinnerSpy, clientReviewService } from 'src/app/helpers/spies';
+
+import { ReviewComponent } from './review.component';
+
+class RouterStub { }
+class ActivatedRouterStub {
+  private subject = new Subject();
+
+  push(value) {
+    this.subject.next(value);
+  }
+  get params() {
+    return this.subject.asObservable();
+  }
+}
+
+describe('ReviewComponent', () => {
+  let component: ReviewComponent;
+  let fixture: ComponentFixture<ReviewComponent>;
+  const mockRouter = routerSpy;
+  const mockToastr = toastServiceSpy;
+  const mockSpinner = spinnerSpy;
+  const mockreviewService = clientReviewService;
+  const mockactivatedRoute = routerSpy;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ReviewComponent],
+      imports: [
+        NgxSpinnerModule, ReactiveFormsModule, FormsModule,
+        HttpClientTestingModule, ToastrModule.forRoot(), BrowserAnimationsModule
+      ],
+      providers: [
+        { provide: Router, useClass: RouterStub },
```

Comment:
Hey @Niyitangasam , I believe this should be
```js
 {provide: Router, useClass: routerSpy}
```
we are mocking the Router class with the `routerSpy` that you have imported, the same should apply for `ActivatedRoute`

Reply:
@Tevinthuku, they are difference between the spies defined in helpers and mocked stubs I used. 
As you suggested to use
 ```js
 {provide: Router, useClass: routerSpy}
```
this can't work, maybe you were suggesting to use ```useValue``` instead of ``` useClass``` like this  ```js
 {provide: Router, useValue: routerSpy}
```
But also this won't work in my case because I did that to mock params in different way